internal/statesync/syncer.go

```diff
@@ -141,7 +141,13 @@ func (s *syncer) AddSnapshot(peerID types.NodeID, snapshot *snapshot) (bool, err
 
 // AddPeer adds a peer to the pool. For now we just keep it simple and send a
 // single request to discover snapshots, later we may want to do retries and stuff.
-func (s *syncer) AddPeer(peerID types.NodeID) {
+func (s *syncer) AddPeer(peerID types.NodeID) (err error) {
+	defer func() {
```

Comment:
Could this please also get a comment for later cleanup? It might qualify as a subcase of #6996.